src/Editor/index.js

```diff
@@ -88,6 +90,24 @@ export class Editor extends React.Component {
         }
     }
 
+    buildMentionsMap(text) {
+      const mentions = EU.findMentions(text)
+      let plainLength = 0
+      let lastIdx = 0
+      mentions.forEach(mention => {
+        plainLength += mention.start - lastIdx
+        lastIdx = mention.end + 1
+        this.mentionsMap.set(
+          [plainLength, plainLength + (mention.username.length)],
```

Comment:
This does not seem to be correct, as in https://github.com/Sircrab/react-native-mentions-editor/blob/54f0c18cd961e26b4bfa61660779c537fd986b7a/src/Editor/index.js#L257-L259 they set the mention to be keyed by the start and end of the mention's position.

Also, it seems to be quite expensive to perform the update of the mentions every time a new character gets added.

Reply:
Originally I had it as you mention, however the `menStartIndex` and `menEndIndex` refer to the indices of the start and ending of the mention in the "plain" text form (the one that you see rendered). The `mention.start` and `mention.end` indices which appear in the `buildMentionsMap` refer to the indices in the "full" text which is the text with all of the mention's info (e.g: `@[me](id: 000000-000000-00000`), hence why I go through the trouble of calculating the indices of the plain text for the mentions.

As for the update of the mentions, it is unfortunately necessary when you begin writing text that is behind an already existing mention (the indices need to be updated to correctly reflect the mention's new position)